### ISSUE TITLE
Not only running on specified path when linting with `mypy`

### ISSUE THREAD

**Issue by amyheather** [lintquarto/lintquarto#41]
As a test case, I tried running `mypy` on the vidigi docs, and the results included:

* Files from the `vidigi/` folder (rather than `vidigi_docs/`)
* Filenames from `vidigi_docs/` showing as `.py` rather than `.qmd` (as covered in #44)

```
(vidigi) amy@xps:~/Documents/general/vidigi$ lintquarto -l mypy -p vidigi_docs
=============================================================
Running mypy...
=============================================================
Success: no issues found in 1 source file
vidigi/resources.py:351: error: Incompatible types in assignment (expression has type "type[PriorityGetLegacy] | MethodType", base class "Store" defined the type as "Callable[[], StoreGet]")  [assignment]
vidigi/prep.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
examples/example_2_branching_multistep/ex_2_model_classes.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
examples/example_2_branching_multistep/ex_2_model_classes.py:5: error: Skipping analyzing "sim_tools.distributions": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:5: error: Skipping analyzing "plotly.graph_objects": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:784: error: Incompatible default for argument "relative_text_position_x" (default has type "float", argument has type "int")  [assignment]
vidigi/animation.py:785: error: Incompatible default for argument "relative_text_position_y" (default has type "float", argument has type "int")  [assignment]
vidigi_docs/vidigi_vs_bupar.py:50: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi_docs/vidigi_vs_bupar.py:50: note: Hint: "python3 -m pip install pandas-stubs"
vidigi_docs/vidigi_vs_bupar.py:50: note: (or run "mypy --install-types" to install all missing stub packages)
vidigi_docs/vidigi_vs_bupar.py:411: error: Skipping analyzing "plotly.io": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi_docs/vidigi_vs_bupar.py:411: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
vidigi_docs/vidigi_vs_bupar.py:411: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
Found 13 errors in 5 files (checked 1 source file)
vidigi_docs/populating_event_logs.py:47: error: Name "event_log" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:48: error: Name "unique_entity_identifier" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:52: error: Name "env" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:68: error: Name "event_log" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:69: error: Name "unique_entity_identifier" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:73: error: Name "env" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:100: error: Name "event_log" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:101: error: Name "unique_entity_identifier" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:105: error: Name "self" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:126: error: "yield" outside function
vidigi_docs/populating_event_logs.py:126: error: Name "my_resource_store" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:128: error: Name "event_log" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:129: error: Name "unique_entity_identifier" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:133: error: Name "env" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:134: error: Name "triage_resource" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:148: error: Name "event_log" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:149: error: Name "unique_entity_identifier" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:153: error: Name "env" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:154: error: Name "triage_resource" is not defined  [name-defined]
vidigi_docs/populating_event_logs.py:158: error: Name "my_resource_store" is not defined  [name-defined]
Found 20 errors in 1 file (errors prevented further checking)
Success: no issues found in 1 source file
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure_old.py:77: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
Success: no issues found in 1 source file
vidigi_docs/introduction.py:388: error: unexpected indent  [syntax]
Found 1 error in 1 file (errors prevented further checking)
vidigi/prep.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:5: error: Skipping analyzing "plotly.graph_objects": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:784: error: Incompatible default for argument "relative_text_position_x" (default has type "float", argument has type "int")  [assignment]
vidigi/animation.py:785: error: Incompatible default for argument "relative_text_position_y" (default has type "float", argument has type "int")  [assignment]
vidigi/utils.py:1: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/utils.py:90: error: Cannot find implementation or library stub for module named "streamlit_javascript"  [import-not-found]
vidigi/logging.py:4: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/logging.py:8: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/logging.py:8: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/logging.py:123: error: Incompatible default for argument "run_number" (default has type "None", argument has type "int")  [assignment]
vidigi/logging.py:123: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
vidigi/logging.py:123: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
vidigi/logging.py:321: error: Function "builtins.any" is not valid as a type  [valid-type]
vidigi/logging.py:321: note: Perhaps you meant "typing.Any" instead of "any"?
vidigi_docs/basic_example_with_vidigi_logger_helpers.py:17: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi_docs/basic_example_with_vidigi_logger_helpers.py:17: note: Hint: "python3 -m pip install pandas-stubs"
vidigi_docs/basic_example_with_vidigi_logger_helpers.py:17: note: (or run "mypy --install-types" to install all missing stub packages)
vidigi_docs/basic_example_with_vidigi_logger_helpers.py:17: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 15 errors in 5 files (checked 1 source file)
vidigi/resources.py:351: error: Incompatible types in assignment (expression has type "type[PriorityGetLegacy] | MethodType", base class "Store" defined the type as "Callable[[], StoreGet]")  [assignment]
vidigi/prep.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:5: error: Skipping analyzing "plotly.graph_objects": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:784: error: Incompatible default for argument "relative_text_position_x" (default has type "float", argument has type "int")  [assignment]
vidigi/animation.py:785: error: Incompatible default for argument "relative_text_position_y" (default has type "float", argument has type "int")  [assignment]
vidigi/utils.py:1: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/utils.py:90: error: Cannot find implementation or library stub for module named "streamlit_javascript"  [import-not-found]
vidigi/logging.py:4: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/logging.py:8: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/logging.py:8: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/logging.py:123: error: Incompatible default for argument "run_number" (default has type "None", argument has type "int")  [assignment]
vidigi/logging.py:123: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
vidigi/logging.py:123: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
vidigi/logging.py:321: error: Function "builtins.any" is not valid as a type  [valid-type]
vidigi/logging.py:321: note: Perhaps you meant "typing.Any" instead of "any"?
vidigi_docs/basic_example_with_vidigi_logger_helpers_multiple_servers.py:21: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi_docs/basic_example_with_vidigi_logger_helpers_multiple_servers.py:21: note: Hint: "python3 -m pip install pandas-stubs"
vidigi_docs/basic_example_with_vidigi_logger_helpers_multiple_servers.py:21: note: (or run "mypy --install-types" to install all missing stub packages)
vidigi_docs/basic_example_with_vidigi_logger_helpers_multiple_servers.py:21: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 16 errors in 6 files (checked 1 source file)
vidigi/prep.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:5: error: Skipping analyzing "plotly.graph_objects": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:784: error: Incompatible default for argument "relative_text_position_x" (default has type "float", argument has type "int")  [assignment]
vidigi/animation.py:785: error: Incompatible default for argument "relative_text_position_y" (default has type "float", argument has type "int")  [assignment]
vidigi/utils.py:1: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/utils.py:90: error: Cannot find implementation or library stub for module named "streamlit_javascript"  [import-not-found]
vidigi_docs/basic_example_minimal_required_vidigi_functions.py:21: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi_docs/basic_example_minimal_required_vidigi_functions.py:21: note: Hint: "python3 -m pip install pandas-stubs"
vidigi_docs/basic_example_minimal_required_vidigi_functions.py:21: note: (or run "mypy --install-types" to install all missing stub packages)
vidigi_docs/basic_example_minimal_required_vidigi_functions.py:21: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 10 errors in 4 files (checked 1 source file)
vidigi/resources.py:351: error: Incompatible types in assignment (expression has type "type[PriorityGetLegacy] | MethodType", base class "Store" defined the type as "Callable[[], StoreGet]")  [assignment]
vidigi/prep.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:3: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:4: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:5: error: Skipping analyzing "plotly.graph_objects": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/animation.py:784: error: Incompatible default for argument "relative_text_position_x" (default has type "float", argument has type "int")  [assignment]
vidigi/animation.py:785: error: Incompatible default for argument "relative_text_position_y" (default has type "float", argument has type "int")  [assignment]
vidigi/utils.py:1: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/utils.py:90: error: Cannot find implementation or library stub for module named "streamlit_javascript"  [import-not-found]
vidigi/logging.py:4: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi/logging.py:8: error: Skipping analyzing "plotly.express": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/logging.py:8: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi/logging.py:123: error: Incompatible default for argument "run_number" (default has type "None", argument has type "int")  [assignment]
vidigi/logging.py:123: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
vidigi/logging.py:123: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
vidigi/logging.py:321: error: Function "builtins.any" is not valid as a type  [valid-type]
vidigi/logging.py:321: note: Perhaps you meant "typing.Any" instead of "any"?
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:18: error: Library stubs not installed for "pandas"  [import-untyped]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:18: note: Hint: "python3 -m pip install pandas-stubs"
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:18: note: (or run "mypy --install-types" to install all missing stub packages)
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:19: error: Skipping analyzing "plotly.io": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:19: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:19: error: Skipping analyzing "plotly": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:124: error: Name "__init__" already defined on line 107  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:316: error: Skipping analyzing "sim_tools.distributions": module is installed, but missing library stubs or py.typed marker  [import-untyped]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:369: error: Name "__init__" already defined on line 265  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:412: error: Name "__init__" already defined on line 265  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:495: error: Name "init_resources" already defined on line 477  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:519: error: Name "self" is not defined  [name-defined]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:521: error: "yield" outside function
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:531: error: Name "self" is not defined  [name-defined]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:533: error: "yield" outside function
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:618: error: Name "attend_clinic" already defined on line 584  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:710: error: Name "__init__" already defined on line 265  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:729: error: Name "__init__" already defined on line 265  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:782: error: Name "run_trial" already defined on line 753  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:983: error: Name "g" already defined on line 86  [no-redef]
vidigi_docs/adding_vidigi_to_a_simple_simpy_model_hsma_structure.py:996: error: Name "Patient" already defined on line 106  [no-redef]
Found 33 errors in 6 files (errors prevented further checking)
```

**Comment by amyheather** [comment]
Filename issue resolved in another commit.

Returning to the other issue - inclusion of files from `vidigi/.`, `vidigi/examples/`, etc. - when it should just be running on `vidigi/vidigi_docs/`.

**Comment by amyheather** [comment]
Have realised this is not an issue but a feature of `mypy` - from the [documentation](https://mypy.readthedocs.io/en/latest/running_mypy.html):

> Mypy is designed to [doggedly follow all imports](https://mypy.readthedocs.io/en/latest/running_mypy.html#finding-imports), even if the imported module is not a file you explicitly wanted mypy to check.

Hence, there is nothing to change, but can make a note in the documentation to expect this behaviour.
